### ISSUE TITLE
binding Java methods in 9.3 breaks compatibility

### ISSUE THREAD

**Issue by kares** [jruby/jruby#7244]
seems the logic should consider using the public abstract super method instead of trying to access the impl directly in cases such as `java.util.concurrent.ConcurrentHashMap` returning a non-public view class: 
`Java::JavaUtilConcurrent::ConcurrentHashMap::KeySetView`

### working in 9.2

`jruby --disable=did_you_mean -v -e "p java.util.concurrent.ConcurrentHashMap.new.keySet.size"`
 ```
jruby 9.2.20.0 (2.5.8) 2021-11-02 1a3255440b OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +jit [linux-x86_64]
0
```

`jruby --disable=did_you_mean -v -e "p java.util.concurrent.ConcurrentHashMap.new.keySet.toArray"`
```
jruby 9.2.20.0 (2.5.8) 2021-11-02 1a3255440b OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +jit [linux-x86_64]
java.lang.Object[]@d70e9a

```

### broken since 9.3

tested using the current jruby-9.3 branch but the regression is present in 9.3.4.0 as well

`bin/jruby --disable=did_you_mean -v -e "p java.util.concurrent.ConcurrentHashMap.new.keySet.size"`
```
jruby 9.3.5.0-SNAPSHOT (2.6.8) 2022-06-07 33d1fa27de OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +jit [x86_64-linux]
TypeError: illegal access on 'size': class org.jruby.javasupport.JavaMethod (in module org.jruby.dist) cannot access a member of class java.util.concurrent.ConcurrentHashMap$CollectionView (in module java.base) with modifiers "public final"
  <main> at -e:1
```

`bin/jruby --disable=did_you_mean -v -e "p java.util.concurrent.ConcurrentHashMap.new.keySet.toArray"`
```
jruby 9.3.5.0-SNAPSHOT (2.6.8) 2022-06-07 33d1fa27de OpenJDK 64-Bit Server VM 11.0.14.1+1 on 11.0.14.1+1 +jit [x86_64-linux]
TypeError: illegal access on 'toArray': class org.jruby.javasupport.JavaMethod (in module org.jruby.dist) cannot access a member of class java.util.concurrent.ConcurrentHashMap$CollectionView (in module java.base) with modifiers "public final"
  <main> at -e:1
```

**Comment by kares** [comment]
confirmed the issue is coming due the fix for #6197 (introduced in JRuby 9.3.0)
particularly, the logic that binds [a `public` method (even if the class is not public)](https://github.com/jruby/jruby/commit/1ba99804552eda599c71494363392dee426229e3#diff-78e2e22601628552e67b4402d044c2ec946e98987b5279f6287675586408aa51R173) seems to be the 'cause'.

still need to figure out the behavioral details as seemingly some cases that I thought would fail do not fail ... e.g. `p java.util.Collections.emptyList().size`

**Comment by kares** [comment]
seems that `JRUBY_OPTS="-J--add-opens=java.base/java.util.concurrent=org.jruby.dist"` serves as a work-around:

`JRUBY_OPTS="-J--add-opens=java.base/java.util.concurrent=org.jruby.dist" bin/jruby "p java.util.concurrent.ConcurrentHashMap.new.keySet.size"` no longer fails!

for the non-modular embedded use-case this would translate to something like: `--add-opens=java.base/java.util.concurrent=ALL-UNNAMED`
